### PR TITLE
added share download print functionality for service order

### PR DESCRIPTION
**Pull request by JKnezovic**

### CODE REVIEW

File: components/Inventory/Dialogs/DeleteDialog.js [hunk 1]
```diff
@@ -2,22 +2,19 @@ import { Dialog, Button } from "react-native-paper";
...
 
-const DeleteDialog = ({ doDelete, setIsDelete, item }) => {
+const DeleteDialog = ({ deleteServiceHistory, setIsDelete, serviceDescription, updateId }) => {
   return (
     <>
       <Dialog.Title>Delete</Dialog.Title>
       <Dialog.Content>
         <Text>{"Are you sure you want to delete:"}</Text>
-        <Text>{item + "?"}</Text>
+        <Text>{serviceDescription + "?"}</Text>
```
Comment (katarina-cudina) [inline components/Inventory/Dialogs/DeleteDialog.js:11]: Nebitno za kod, al mislim da ovdi možemo uklonit dvotočku

File: components/OrderDetails/PrintShare.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  const printToFile = async () => {
+    const htmlContent = html_template(service, serviceHistory, orderStatuses);
+    const { uri } = await Print.printToFileAsync({ html: htmlContent });
+    await shareAsync(uri, { UTI: ".pdf", mimeType: "application/pdf" });
+  };
+
+  return (
+    <View style={styles.container}>
+      <AntDesign name="printer" size={28} color="black" onPress={print} style={styles.icon} />
```
Comment (katarina-cudina) [inline components/OrderDetails/PrintShare.js:26]: Ček ovdi imamo jednu ikonu za print a drugu printToFile? Jel to kao saveToFile?

File: components/OrderDetails/hooks/useGetServicesHistory.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    const serviceObject = new Parse.Object("Services", {
+      id: service.serviceOrderId,
+    });
+    query.equalTo("service_fkey", serviceObject);
+    try {
+      const result = await query.find();
+      setServiceHistory(result);
+    } catch (err) {
+      setError(err.message);
+    } finally {
```
Comment (katarina-cudina) [inline components/OrderDetails/hooks/useGetServicesHistory.js:22]: hot

**Review by katarina-cudina — APPROVED** [review]
gas

**Comment by katarina-cudina** [comment]
Ipak sam našla mali problem, ako klikneš na Print ikonu, ali nakon toga ne isprintaš nego stisneš Cancel, dobiješ Possible unhandled promise rejection. Mislim da to nije error al moga bi se riješit taj warning tako da imamo to handlano isto
Screenshot 2024-11-02 at 13 52 48

**Comment by katarina-cudina** [comment]
Simulator Screenshot - iPad Air (5th generation) - 2024-11-02 at 13 54 30

I jel mora stat ovi bubble dialog ovako, jel može po sredini il bliže botunu il nešto

File: components/OrderDetails/PrintShare.js (hunk 2 above)
Comment (katarina-cudina) [reply components/OrderDetails/PrintShare.js:26]: Mislim da bi se ova druga mogla zvat shareAsPdf?

File: components/Inventory/Dialogs/DeleteDialog.js (hunk 1 above)
Comment (JKnezovic) [reply components/Inventory/Dialogs/DeleteDialog.js:11]: done

File: components/OrderDetails/PrintShare.js (hunk 2 above)
Comment (JKnezovic) [reply components/OrderDetails/PrintShare.js:30]: done

**Comment by JKnezovic** [comment]
> Ipak sam našla mali problem, ako klikneš na Print ikonu, ali nakon toga ne isprintaš nego stisneš Cancel, dobiješ Possible unhandled promise rejection. Mislim da to nije error al moga bi se riješit taj warning tako da imamo to handlano isto Screenshot 2024-11-02 at 13 52 48

Note: on iOS, when printing without providing a PrintOptions.printerUrl the Promise will be resolved once printing is started in the native print window and rejected if the window is closed without starting the print. On Android the Promise will be resolved immediately after displaying the native print window and won't be rejected if the window is closed without starting the print.

**Pull request merged**